doc/en/user/source/rest/api/stylegenerator.rst

```diff
@@ -0,0 +1,125 @@
+.. _rest_api_stylegenerator:
+
+Style Generator
+===============
+
+The Style Generator endpoint is used for generating default styles. Styles generated using this endpoint may be added to geoserver using the :ref:`styles <rest_api_styles>` endpoint.
+
+``/stylegenerator[.<format>]``
+------------------------------
```

Comment:
@jodygarnett - I looked at the code / documentation and PUT is already implemented - see [/styles/\<s\>\[.\<format\>\]](http://docs.geoserver.org/stable/en/user/rest/api/styles.html#styles-s-format). I am not sure why you thought it wasn't implemented? Wherever we know the style name, PUT is used, otherwise POST is used, in accordance with REST conventions.